plugins/pantheon-files-ctags/plugin.vala

```diff
@@ -328,180 +328,151 @@ public class Marlin.Plugins.CTags : Marlin.Plugins.Base {
         }
     }
 
-    private class ColorWidget : Gtk.MenuItem {
-        private new bool has_focus;
-        private const int BUTTON_WIDTH = 10;
-        private const int BUTTON_HEIGHT = 10;
-        /* Set start margin to match other menuitems. Is there a way to determine
-         * this programmatically?  */
-        private const int MARGIN_START = 27;
-        private const int SPACING = 15;
+    private class ColorButton : Gtk.Button {
```

Comment:
You need to give each button an id property (just an int will do) so that the shared signal handler can determine the source button.

Reply:
So in the main public method of the custom class I add in a `int id`, and assign one for each button?